Script/update_grovepi.sh

```diff
@@ -5,34 +5,147 @@ PIHOME=/home/pi
 DEXTER=Dexter
 DEXTER_PATH=$PIHOME/$DEXTER
 RASPBIAN=$PIHOME/di_update/Raspbian_For_Robots
-curl --silent https://raw.githubusercontent.com/DexterInd/script_tools/master/install_script_tools.sh | bash
+GROVEPI_DIR=$DEXTER_PATH/GrovePi
+DEXTERSCRIPT=$DEXTER_PATH/lib/Dexter/script_tools
 
-# needs to be sourced from here when we call this as a standalone
-source /home/pi/$DEXTER/lib/$DEXTER/script_tools/functions_library.sh
+# whether to install the dependencies or not (avrdude, apt-get, wiringpi, and so on)
+installdependencies=true
+updaterepo=true
+install_pkg_scriptools=true
 
-GROVEPI_DIR=$DEXTER_PATH/GrovePi
+# the following 3 options are mutually exclusive
+systemwide=true
+userlocal=false
+envlocal=false
+usepython3exec=true
+
+# the following option tells which branch has to be used
+selectedbranch="master"
 
-# Check for a GrovePi directory under "Dexter" folder.  If it doesn't exist, create it.
+declare -a optionslist=("--system-wide")
+# iterate through bash arguments
+for i; do
+  case "$i" in
+    --no-dependencies)
+      installdependencies=false
+      ;;
+    --no-update-aptget)
+      updaterepo=false
+      ;;
+    --bypass-pkg-scriptools)
+      install_pkg_scriptools=false
+      ;;
+    --user-local)
+      userlocal=true
+      systemwide=false
+      declare -a optionslist=("--user-local")
+      ;;
+    --env-local)
+      envlocal=true
+      systemwide=false
+      declare -a optionslist=("--env-local")
+      ;;
+    --system-wide)
+      ;;
+    develop|feature/*|hotfix/*|fix/*|DexterOS*|v*)
+      selectedbranch="$i"
+      ;;
+  esac
+done
 
-if [ -d "$GROVEPI_DIR" ]; then
-    echo "GrovePi Directory Exists"
-    cd $GROVEPI_DIR             # Go to directory
-    sudo git fetch origin       # Hard reset the git files
-    sudo git reset --hard  
-    sudo git merge origin/master
+# show some feedback for the GrovePi
```

Comment:
Please keep this branding in a function so it can be out of sight, out of mind when we work in the script.

Reply:
I can put it in a file separate other than this one and the `install.sh` script.